dev-docs/security-overview.md

```diff
@@ -0,0 +1,417 @@
+
+1. The `nydus-snapshotter` plugin in `containerd` fetches container metadata and forwards it to the `kata-agent` inside the CVM.
+2. `image-rs`, running alongside the `kata-agent`, pulls the container images and computes their hashes. The `kata-agent` then verifies these computed hashes against those specified in the security policy.
+3. If the hashes match, the `kata-agent` starts the container workload inside the CVM.
+
+✅ After completing these steps, the workload runs securely within a confidential environment.
+
+### 5️⃣ Contrast-specific containers on worker nodes
+
+Each CVM pod includes two additional containers that start before the primary application container:
+
+#### **Init container**
+
+- Runs to completion before the main application container starts.
+- Obtains a service mesh certificate, enabling the pod to join the internal service mesh managed by the Contrast coordinator.
+- Its integrity is enforced and verified according to the security policy.
+
+#### **Sidecar container**
+
+- Runs continuously alongside the application container, functioning as a network proxy.
+- Encrypts all pod network traffic using mutual TLS (mTLS) based on certificates provided by the service mesh.
+- Activated via annotations in the deployment YAML.
+- Ensures all traffic passes through the proxy by configuring `iptables` rules.
+- Uses [**Envoy**](https://www.envoyproxy.io/) to manage mTLS encryption for both incoming (ingress) and outgoing (egress) traffic.
+  - Employs [TPROXY](https://docs.kernel.org/networking/tproxy.html) for handling ingress traffic.
+- Its integrity is similarly enforced through the security policy.
+- For more details, see [the service mesh documentation](https://docs.edgeless.systems/contrast/components/service-mesh#ingress).
+
+✅ This setup ensures that the CVM can be reliably attested and that all communications remain end-to-end encrypted.
+
+### 6️⃣ Workload entry into the Contrast service mesh
+
+#### 1. The Init container inside the CVM pod generates a cryptographic key pair (private and public).
+
+- The private key remains securely inside the CVM and is never exposed externally.
+
+#### 2. The Init container requests a service mesh certificate from the Contrast coordinator via rRPC, providing its public key.
+
+#### 3. The Contrast coordinator (acting as a client) establishes an attested TLS (aTLS) connection with the Init container of the CVM pod (server) on the worker node:
+
+- The TLS protocol is extended to include attestation verification of the server.
+- The worker node must successfully pass attestation against the defined manifest before the TLS connection is fully established.
+- Only after successful attestation does the coordinator issue a valid service mesh certificate.
+
+#### 4. The coordinator securely delivers the service mesh certificate (signed by the Intermediate CA) to the Init container.
+
+- Additionally, it provides a secret seed used for key recovery purposes.
```

Comment:
The "secret seed" is what the Coordinator itself works with, something like a master key for the entire cluster. The initializers only receive a "workload secret", which is stable for a given fixed pod definition and can be used to encrypt and persist.